pkg/resource/resource.go

```diff
@@ -283,6 +283,26 @@ func MustBeControllableBy(u types.UID) ApplyOption {
 	}
 }
 
+// MustBeControllableByDesiredController requires that the current object is
+// controllable by the desired object's controller. An object is controllable if
+// its controller reference matches desired object's controller's UID, or it has
+// no controller reference.
+func MustBeControllableByDesiredController() ApplyOption {
```

Comment:
I was hoping to remove this pattern in which our required controller is implicitly inferred from the desired object when I deprecated `ControllersMustMatch`. Is it possible to use `ControllabeBy` instead?

Reply:
@negz It could be removed, but it would result in the user never being to override this in the `Workload` reconciler. So all `ApplyOptions` provided would be in addition to `MustBeControllableBy` (which I am assuming is what you are referring to as `ControllableBy`). Do you prefer that?

Reply:
I prefer that, at least for the time being.